source/mir/random/tinflex/internal/area.d

```diff
@@ -0,0 +1,700 @@
+module mir.random.tinflex.internal.area;
+
+import mir.random.tinflex.internal.types : Interval;
+import std.traits : ReturnType;
+
+/**
+Determines the hat and squeeze function of an interval.
+Based on Theorem 1
+*/
+void determineSqueezeAndHat(S)(ref Interval!S iv)
+in
+{
+    assert(iv.lx < iv.rx, "invalid interval");
```

Comment:
> Isn't this something that could be guaranteed by Interval ... ?

It wasn't, but it makes a lot of sense to check this in the invariant. Thanks for the suggestion!